src/sinks/pulsar.rs

```diff
@@ -84,21 +82,27 @@ impl SinkConfig for PulsarSinkConfig {
 }
 
 impl PulsarSink {
-    pub(crate) fn new(config: PulsarSinkConfig, acker: Acker) -> crate::Result<Self> {
-        let auth = config.auth.map(|auth| Authentication {
-            name: auth.name,
+    fn new(config: &PulsarSinkConfig, acker: Acker) -> crate::Result<Self> {
+        // Clone for moving to async lifetime
+        let address = config.address.clone();
+        let auth = config.auth.as_ref().map(|auth| Authentication {
+            name: auth.name.clone(),
             data: auth.token.as_bytes().to_vec(),
         });
-        let pulsar = Pulsar::new(config.address.parse()?, auth, DefaultExecutor::current())
-            .wait()
+        let topic = config.topic.clone();
+
+        // Temporary async block in sync
+        let mut rt = tokio_compat::runtime::current_thread::Runtime::new()?;
+        let producer = rt
+            .block_on_std(async move {
+                let pulsar = Pulsar::new(&address, auth, None, None).await?;
```

Comment:
We create `Pulsar` here in temporary runtime and this bind this new object to temporary runtime... when runtime dropped we will not able communicate with pulsar.